File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/1258/Camisetas.java
import java.util.*;

class Camisa{
    String nome;
    String cor;
    char tamanho;

    public Camisa(String nome, String cor, char tamanho){
        this.nome = nome;
        this.cor = cor;
        this.tamanho = tamanho;
    }
}

class Lista{
    private Camisa[] camisas;
    private int n;

    public Lista(){
        camisas = new Camisa[1000];
        n = 0;
    }

    public void inserir(Camisa c) throws Exception{
        if(n >= camisas.length){
            throw new Exception("Erro ao inserir!");
        }
        camisas[n++] = c;
    }

    public Camisa get(int i){
        return camisas[i];
    }

    public int tamanho(){
        return n;
    }

    public void ordenar(){
        for(int i = 0; i < n - 1; i++){
            for(int j = 0; j < n - i - 1; j++){
                int cmpCor = camisas[j].cor.compareTo(camisas[j + 1].cor);
                if(cmpCor > 0 ||
                    (cmpCor == 0 && camisas[j].tamanho < camisas[j + 1].tamanho) || 
                    (cmpCor == 0 && camisas[j].tamanho == camisas[j + 1].tamanho &&
                     camisas[j].nome.compareTo(camisas[j + 1].nome) > 0)){

                    Camisa temp = camisas[j];
                    camisas[j] = camisas[j + 1];
                    camisas[j + 1] = temp;
                }
            }
        }
    }
}

public class Camisetas{
    public static void main(String[] args) throws Exception{
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        sc.nextLine();

        while(N != 0){
            Lista lista = new Lista();

            for(int i = 0; i < N; i++){
                String nome = sc.nextLine();
                String cor = sc.next();
                char tamanho = sc.next().charAt(0);
                sc.nextLine();

                Camisa c = new Camisa(nome, cor, tamanho);
                lista.inserir(c);
            }

            lista.ordenar();

            for(int i = 0; i < lista.tamanho(); i++){
                Camisa c = lista.get(i);
                System.out.printf("%s %c %s\n", c.cor, c.tamanho, c.nome);
            }
            System.out.println();
            N = sc.nextInt();
            sc.nextLine();
        }

        sc.close();
    }
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/beecrowd/1258/Camisetas.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct{
    char nome[100];
    char cor[100];
    char tamanho;
} Camisa;

typedef struct{
    Camisa camisas[1000];
    int n;
} Lista;

void inserir(Lista *lista, Camisa c){
    if(lista->n >= 1000){
        printf("Erro ao inserir!\n");
        exit(1);
    }
    lista->camisas[lista->n++] = c;
}

Camisa get(Lista *lista, int i){
    return lista->camisas[i];
}

int tamanho(Lista *lista){
    return lista->n;
}

void ordenar(Lista *lista){
    for(int i = 0; i < lista->n - 1; i++){
        for(int j = 0; j < lista->n - i - 1; j++){
            Camisa *a = &lista->camisas[j];
            Camisa *b = &lista->camisas[j + 1];

            int cmpCor = strcmp(a->cor, b->cor);
            if(cmpCor > 0 ||
                (cmpCor == 0 && a->tamanho < b->tamanho) ||
                (cmpCor == 0 && a->tamanho == b->tamanho &&
                 strcmp(a->nome, b->nome) > 0)){
                Camisa temp = *a;
                *a = *b;
                *b = temp;
            }
        }
    }
}

int main(){
    int N;
    scanf("%d", &N);
    getchar();

    while(N != 0){
        Lista lista;
        lista.n = 0;

        for(int i = 0; i < N; i++){
            Camisa c;
            fgets(c.nome, 100, stdin);
            c.nome[strcspn(c.nome, "\n")] = '\0';

            scanf("%s %c", c.cor, &c.tamanho);
            getchar();

            inserir(&lista, c);
        }

        ordenar(&lista);

        for(int i = 0; i < tamanho(&lista); i++){
            Camisa c = get(&lista, i);
            printf("%s %c %s\n", c.cor, c.tamanho, c.nome);
        }

        printf("\n");
        scanf("%d", &N);
        getchar();
    }

    return 0;
}